### PR TITLE
[Bugfix] Fix scrolling and fetch issues.

### PR DESCRIPTION
**Pull request by palatter**
## Purpose

* ...

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

**Comment by kevinyulu** [comment]
I noticed that the loading spinner appears at the bottom of the chat view, everything else looks good.

**Comment by palatter** [comment]
> I noticed that the loading spinner appears at the bottom of the chat view, everything else looks good. Simulator Screen Shot - iPhone 13 Pro - 2022-12-21 at 15 52 26

Fixed.

**Comment by JoshuaLai** [comment]
Tested and seems to be working as expected for small chat and long chat. Known issue related with the spinner now disappearing at the top of the chat if we have fetched all messages. There also seem to be some swift lint issues in the branch that need to be resolved.

**Review by JoshuaLai — DISMISSED** [review]

**Comment by Leah-Xia-Microsoft** [comment]
Seems like when it reaches the first message of the thread, the fetch activity indicator doesn't disappear.
https://user-images.githubusercontent.com/107075081/210020762-c583d72a-b97a-4872-93be-aa2e8b31dc6f.MP4

**Review by Leah-Xia-Microsoft — CHANGES_REQUESTED** [review]
2 edge cases for fetch. Everything else looks awesome.

File: AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift [hunk 1]
```diff
@@ -68,8 +70,11 @@ class MessageListViewModel: ObservableObject {
         shouldScrollToBottom = true
     }
 
-    func fetchMessages(index: Int) {
-        if index == minFetchIndex {
+    func fetchMessages(lastSeenMessage: ChatMessageInfoModel) {
+        if let lastSeenMessageIndex = messages.firstIndex(where: { $0.id == lastSeenMessage.id }),
+            lastSeenMessageIndex <= minFetchIndex && hasFetchedPreviousMessages {
+            print("SCROLL: Fetch Messages")
```
Comment (Leah-Xia-Microsoft) [inline AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift:76]: Do we intend to keep this print?

Comment (palatter) [reply AzureCommunicationUI/sdk/AzureCommunicationUIChat/Sources/Presentation/SwiftUI/Chat/ChatViewComponents/MessageListViewModel.swift:76]: Oh good catch! I think there are a couple of those. I'll remove.

**Comment by palatter** [comment]
> Seems like when it reaches the first message of the thread, the fetch activity indicator doesn't disappear. https://user-images.githubusercontent.com/107075081/210020762-c583d72a-b97a-4872-93be-aa2e8b31dc6f.MP4
> 
> Also, when leaving the thread and reenter, fetch indicator doesn't disappear either. For this one, it might be related to
> 
> ```
>     func onDisappear() {
>         self.hasFetchedInitialMessages = false
>         didEndScrollingTimer?.invalidate()
>     }
> ```
> 
>  RPReplay_Final1672356930.MP4

Fixed the issue where the activity indicator was staying on screen when returning.

For the other issue, I want to discuss the best approach with Jim, as it will involve determining if we have gotten all messages in the wrapper/service level. So I'll leave that for another PR.

**Review by JoshuaLai — DISMISSED** [review]

**Review by Leah-Xia-Microsoft — APPROVED** [review]
The send read receipt feature is affected but I'll fix it in another PR

**Review by JoshuaLai — APPROVED** [review]

**Pull request merged**